templates/build-preview.yml

```
@@ -0,0 +1,29 @@
+# File: templates/build-preview.yml
+
+parameters:
+  BuildConfiguration: ''
+  Arguments: ''
+  VmImage: ''
+  OS: 'Windows'
+
+jobs:
+
+- job:
+  displayName: Preview Flag ${{ parameters.BuildConfiguration }} 
+  pool:
+    vmImage: ${{ parameters.VmImage }}
+
+  steps:
+  - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
+    clean: true  # if true, execute `execute git clean -ffdx && git reset --hard HEAD` before fetching
+
+  - task: DotNetCoreCLI@2
+    displayName: Build Microsoft.Azure.Cosmos PREVIEW
+    inputs: 
+      command: build  
+      configuration: $(parameters.BuildConfiguration)
+      nugetConfigPath: NuGet.config
+      projects: Microsoft.Azure.Cosmos.sln 
+      arguments: -p:Optimize=true -p:DefineConstants="DOCDBCLIENT COSMOSCLIENT NETSTANDARD20 PREVIEW"
```

nit: I think you can just do 
```suggestion
      arguments: -p:Optimize=true -p:IsPreview=true
```

---

That runs `dotnet build -p:IsPreview=true` and passes the `IsPreview` flag that sets the required constants.